source/Mission.cpp

```diff
@@ -829,6 +831,20 @@ bool Mission::Do(Trigger trigger, PlayerInfo &player, UI *ui, const shared_ptr<S
 		--player.Conditions()[name + ": active"];
 		++player.Conditions()[name + ": failed"];
 	}
+	else if(trigger == ABORT)
+	{
+		// Only decrement the active mission condition if an ABORT action
+		// exists. Otherwise, this condition will be decremented when the
+		// FAIL trigger is done.
+		if(hasAction)
```

Comment:
If you accept my change above, then the update of the player.Conditions has to be done always.